graylog2-web-interface/src/views/components/searchbar/QueryInput.tsx

```diff
@@ -62,23 +70,32 @@ const handleExecution = (editor, onExecute, value, error) => {
   onExecute(value);
 };
 
-const _configureEditor = (node, completer, onExecute) => {
-  const editor = node && node.editor;
+const displaySuggestionAfterFieldName = (editor, currentLine: number, lines: Array<Array<Line>>) => {
+  const currentToken = lines[currentLine - 1].find((token) => token?.start !== undefined);
 
-  if (editor) {
-    editor.commands.addCommands([{
-      name: 'Execute',
-      bindKey: 'Enter',
-      exec: onExecute,
-    },
-    {
-      name: 'SuppressShiftEnter',
-      bindKey: 'Shift-Enter',
-      exec: () => {},
-    }]);
+  if (!currentToken) {
+    return;
+  }
 
+  if (currentToken.type === 'keyword' && currentToken.value.endsWith(':')) {
+    editor.execCommand('startAutocomplete');
+  }
+};
+
+const _configureEditor = (editor, completer: AutoCompleter, configuredListeners: React.MutableRefObject<boolean>) => {
+  if (editor) {
     editor.commands.removeCommands(['find', 'indent', 'outdent']);
+    // eslint-disable-next-line no-param-reassign
     editor.completers = [completer];
+
+    if (!configuredListeners.current) {
+      editor.session.on('tokenizerUpdate', (input, { bgTokenizer: { currentLine, lines } }) => {
+        displaySuggestionAfterFieldName(editor, currentLine, lines);
```

Comment:
Maybe we can generalize the logic here so that we do not couple field name completion to the general completion logic. We could do something like this:
```suggestion
        if (completer.shouldShowCompletions(editor, currentLine, lines)) {
          editor.execCommand('startAutocomplete');
        }
```

Where `SearchBarAutoCompletions#shouldShowCompletions` would delegate the call to the different completers and returns if any one of them wants to show suggestions at this point.

Reply:
That is a really good Idea, it worked very well.